src/io/flutter/survey/FlutterSurveyNotifications.java

```diff
@@ -0,0 +1,158 @@
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.jetbrains.lang.dart.DartFileType;
+import icons.FlutterIcons;
+import io.flutter.FlutterInitializer;
+import io.flutter.FlutterMessages;
+import io.flutter.pub.PubRoot;
+import org.jetbrains.annotations.NotNull;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.TimeZone;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+// TODO(pq): remove for M36 / post: 2019/05/27.
+public class FlutterSurveyNotifications {
+  private static final int NOTIFICATION_DELAY_IN_SECS = 3;
+
+  private static final String FLUTTER_LAST_SURVEY_CHECK_KEY = "FLUTTER_LAST_SURVEY_CHECK_KEY";
+  private static final long CHECK_INTERVAL_IN_MS = TimeUnit.HOURS.toMillis(40);
+
+  private static final String SURVEY_TAKEN = "io.flutter.survey.2019.alreadyTaken";
+  private static final String SURVEY_URL = "https://google.qualtrics.com/jfe/form/SV_3W3aVD2y9CoAe6V?Source=IntelliJ";
+
+  private static long SURVEY_START;
+  private static long SURVEY_END;
+
+  static {
+    final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm");
+    dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+    try {
+      SURVEY_START = dateFormat.parse("2019/05/13 19:00").getTime();
+      SURVEY_END = dateFormat.parse("2019/05/27 07:00").getTime();
+    }
+    catch (ParseException e) {
+      // Shouldn't happen.
+    }
+  }
+
+  private static boolean isSurveyOpen() {
+    final long now = System.currentTimeMillis();
+    return now >= SURVEY_START && now <= SURVEY_END;
+  }
+
+  interface FlutterSurveyNotifier {
+    void prompt();
+  }
+
+  public static void init(@NotNull Project project) {
+    project.getMessageBus().connect().subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, new FileEditorManagerListener() {
+      @Override
+      public void fileOpened(@NotNull final FileEditorManager source, @NotNull final VirtualFile file) {
+        check(file);
+      }
+
+      @Override
+      public void selectionChanged(@NotNull FileEditorManagerEvent event) {
+        final VirtualFile file = event.getNewFile();
+        if (file != null) {
+          check(file);
+        }
+      }
+
+      private void check(@NotNull VirtualFile file) {
+        if (PubRoot.isPubspec(file) || file.getFileType() == DartFileType.INSTANCE) {
+          new FlutterSurveyNotifications(project).checkForDisplaySurvey();
+        }
+      }
+    });
+  }
+
+  @NotNull final Project myProject;
+
+  FlutterSurveyNotifications(@NotNull Project project) {
+    this.myProject = project;
+  }
+
+  private void checkForDisplaySurvey() {
+    // Limit display to the survey window.
+    if (!isSurveyOpen()) return;
+
+    final PropertiesComponent properties = PropertiesComponent.getInstance();
+
+    // Don't prompt more often than every 40 hours.
```

Comment:
40 hours seems a bit random. Why not 48, or 168?

Reply:
Heh, that's my fault :D I started using 20 hours for things like this so that they won't show again "the same day", but will tend to appear at the start of the day rather than end if you'd first seen them at the end of the day. We wanted around two days with this one, and I lazily doubled it, and Phil copied it.

No real thought went into it though - if anyone has a preference I don't mind tweaking it in VS Code :-)

Reply:
That's as good a reason as any :)